1a) Uma variável aleatória é uma função que associa um valor numérico a cada resultado de um experimento aleatório.

Discreta: O número de carros vendidos por uma concessionária em um dia é uma variável aleatória discreta, pois pode ser contado e tem valores inteiros (0, 1, 2, ...).

Contínua: A altura de estudantes em uma escola é uma variável aleatória contínua, pois pode assumir qualquer valor dentro de um intervalo e não é limitada a valores contáveis.


1b) Uma distribuição discreta de probabilidade descreve a probabilidade de ocorrência de cada valor possível de uma variável aleatória discreta. As duas condições que ela deve satisfazer são:

Cada probabilidade deve estar entre 0 e 1, inclusive.

A soma de todas as probabilidades deve ser igual a 1²².

1c) O valor esperado de uma distribuição de probabilidade, também conhecido como média ou expectância, não precisa ser um dos valores possíveis que a variável aleatória pode assumir. Ele é calculado como a soma do produto de cada valor possível pela sua probabilidade correspondente. Portanto, o valor esperado é uma média ponderada de todos os resultados possíveis e pode não ser um resultado que a variável aleatória pode de fato assumir em um único experimento. É uma medida de tendência central que dá uma ideia do resultado médio se o experimento for repetido muitas vezes.


In [ ]:
import math

b = 0
c = 0
f = 0


def possion(u, x):
    return ((u ** x) * (math.e ** -u)) / math.factorial(x)


a = possion(5, 5)
print(f'2a) {a}')
for i in range(1, 4):
    b += possion(5, i)
print(f'2b) {b}')

for i in range(0, 5):
    c += possion(5, i)
print(f'2c) {c}')

d = 1 - c - a

print(f'2d) {d}')

print(f'2e) {possion(5, 0)}')

for i in range(3, 7):
    f += possion(5, i)
print(f'2f) {f}')


2a) 0.17546736976785074
2b) 0.25828796829827627
2c) 0.44049328506521246
2d) 0.38403934516693683
2e) 0.006737946999085469
2f) 0.6375314434898577


In [ ]:
import math


def binomial_distribution(chance, tentativas, sucessos):
    return (math.factorial(tentativas) / ((math.factorial(sucessos)) * (math.factorial(tentativas - sucessos)))) * (
            chance ** sucessos) * ((1 - chance) ** (tentativas - sucessos))


print(f'3) {binomial_distribution(0.9, 3, 2)}')


3) 0.24299999999999997


In [ ]:
def geometrical_distribution(chance, tentativa):
    return chance * (1 - chance) ** (tentativa - 1)


print(f'4) {geometrical_distribution(0.75, 3) + geometrical_distribution(0.75, 4)}')


4) 0.05859375


In [ ]:
import math

def possion(u, x):
    return ((u ** x) * (math.e ** -u)) / math.factorial(x)

print(f'5) {1 - possion((0.02*24), 1) - possion((0.02*24), 0)}')

5) 0.07128384673606743


In [ ]:
import math


def probabilidade_falha(taxa, horas):
    return math.exp(-taxa * horas)


print(f"6a) {probabilidade_falha((1 / 5), 2)}")
print(f"6b) {probabilidade_falha((1 / 5), 4) - probabilidade_falha((1 / 5), 6)}")


6a) 0.6703200460356393
6b) 0.14813475220501954


In [ ]:
import math


def binomial_distribution(chance, tentativas, sucessos):
    return (math.factorial(tentativas) / ((math.factorial(sucessos)) * (math.factorial(tentativas - sucessos)))) * (
            chance ** sucessos) * ((1 - chance) ** (tentativas - sucessos))


print(f'7) {binomial_distribution(0.05, 10, 3)}')


7) 0.010475059441406248


In [ ]:
def geometrical_distribution(chance, tentativa):
    return chance * (1 - chance) ** (tentativa - 1)


b = 0
c = 0

print(f'8a) {geometrical_distribution(0.01, 10)}')

for i in range(1, 4):
    b += geometrical_distribution(0.01, i)

print(f'8b) {b}')

for i in range(1, 11):
    c += geometrical_distribution(0.01, i)

print(f'8c) {1 - c}')


8a) 0.009135172474836408
8b) 0.029701
8c) 0.9043820750088045


In [ ]:
def geometrical_distribution(chance, tentativa):
    return chance * (1 - chance) ** (tentativa - 1)


b = 0
c = 0

print(f'9a) {geometrical_distribution(1/4, 4)}')

for i in range(1, 4):
    b += geometrical_distribution(1/4, i)

print(f'9b) {b}')

for i in range(1, 5):
    c += geometrical_distribution(1/4, i)

print(f'9c) {1 - c}')


9a) 0.10546875
9b) 0.578125
9c) 0.31640625


In [ ]:
import math

b = 0
c = 0


def binomial_distribution(chance, tentativas, sucessos):
    return (math.factorial(tentativas) / ((math.factorial(sucessos)) * (math.factorial(tentativas - sucessos)))) * (
            chance ** sucessos) * ((1 - chance) ** (tentativas - sucessos))


print(f'10a) {binomial_distribution(0.29, 7, 2)}')

for i in range(4, 8):
    b += binomial_distribution(0.29, 7, i)

print(f'10b) {b}')

for i in range(1, 5):
    c += binomial_distribution(0.29, 7, i)

print(f'10c) {c}')


10a) 0.31864494568010987
10b) 0.11344235560819994
10c) 0.8842067203047397


In [ ]:
from scipy.integrate import quad


def integrando(x):
    return x / 2


print(f"11) {quad(integrando, 1, 2)[0]}")




---



In [ ]:
def margem_erro(nivel, desvio, amostra):
    return nivel * (desvio / (amostra ** 0.5))


margem = margem_erro(1.645, 1.5, 30)

print(f'12) 22.9 +- {margem}')


12) 22.9 +- 0.4505018035479991


In [ ]:
from scipy.stats import norm


def escore_padrao(valor, media, desvio):
    return (valor - media) / desvio


vinte_mil = escore_padrao(20000, 27989, 3250)
trinta_mil = escore_padrao(30000, 27989, 3250)

print(f'13a) {norm.cdf(trinta_mil) - norm.cdf(vinte_mil)}')
print(f'13b) {norm.cdf(vinte_mil)}')


13a) 0.7249831358225932
13b) 0.006982666701149653


In [ ]:
import math


def probabilidade_falha(taxa, horas):
    return math.exp(-taxa * horas)


print(f"14a) {probabilidade_falha(0.02, 50)}")
print(f"14b) {1 - probabilidade_falha(0.02, 24)}")

14a) 0.36787944117144233
14b) 0.38121660819385916


In [ ]:
from scipy.integrate import quad
import math

def integrando(t):
    return 0.01 * math.e ** (-0.01 * t)


print(f"15) {1 - quad(integrando, 0, 50)[0]}")


15) 0.6065306597126334


a)

$$ P(X=x) = (1-p)^{x-1}p $$


$$ P(X=x) = (0.8)^{x-1} \times 0.2 $$


b)
$$ P(X \geq 3) = 1 - P(X=1) - P(X=2) $$
$$ P(X \geq 3) = 1 - (0.2) - (0.8 \times 0.2) $$
$$ P(X \geq 3) = 1 - 0.2 - 0.16 $$
$$ P(X \geq 3) = 0.64 $$

c)
$$ P(X \text{ é par}) = \sum_{k=1}^{\infty} P(X=2k) $$
$$ P(X \text{ é par}) = \sum_{k=1}^{\infty} (0.8)^{2k-1} \times 0.2 $$
$$ P(X \text{ é par}) = \frac{a}{1-r} $$
$$ P(X \text{ é par}) = \frac{0.16}{1-0.64} $$
$$ P(X \text{ é par}) = \frac{0.16}{0.36} $$
$$ P(X \text{ é par}) = \frac{4}{9} \approx 0.4444 $$